integration_test/README.md

```diff
@@ -1,12 +1,89 @@
-application support. This test is currently implemented as a Kokoro build
-that is run on all PRs. The test builds the Ops Agent on one distro (currently
-Debian 10) and attempts to verify, for each application in
-`supported_applications.txt`, that the application can be installed on a real GCE
-VM and that a single representative metric is successfully uploaded to Google Cloud
-Monitoring.
+Integration tests are implemented as Kokoko builds that run on each PR. The
+builds first build the Ops Agent and then run tests on that agent. The Kokoro
+builds are split up by distro.
+
+## Ops Agent Test
+
+This test exercises "core" features of the Ops Agent such as watching syslog or
+a custom log file. It is implemented in ops_agent_test.go. It can be run outside
+of Kokoro with some setup.
+
+### Setup
+
+You will need a project to run VMs in and a service account with permissions to
+run various operations in the project. You can set up such a service account with
+a few commands:
+
+```
+gcloud iam service-accounts create "service-account-$(whoami)" \
+  --project "${PROJECT}" \
+  --display-name "service-account-$(whoami)"  
+```
+
+```
+for ROLE in \
+    logging.logWriter \
+    monitoring.metricWriter \
+    stackdriver.resourceMetadata.writer \
+    logging.viewer \
+    monitoring.viewer \
+    stackdriver.resourceMetadata.viewer \
+  ; do
+      gcloud projects add-iam-policy-binding "${PROJECT}" \
+          --member "serviceAccount:service-account-$(whoami)@${PROJECT}.iam.gserviceaccount.com" \
+          --role "roles/${ROLE}" > /dev/null
+  done
+```
+
+You will also need a GCS bucket that is used to transfer files onto
+the testing VMs. Store it in env variable `$BUCKET` and add your
+service account as principal with role "Storage Admin" to the bucket:
+
+```
+gsutil iam ch \
+  "serviceAccount:service-account-$(whoami)@${PROJECT}.iam.gserviceaccount.com:roles/storage.admin" \
+  "gs://${BUCKET}"
+```
+
+Finally, download your credentials as a JSON file:
+
+```
+gcloud iam service-accounts keys create $HOME/credentials.json \
+  --project "${PROJECT}" \
+  --iam-account "service-account-$(whoami)@${PROJECT}.iam.gserviceaccount.com"
+```
+
+### Testing Command
+
+When the setup steps are complete, you can run ops_agent_test like this:
+
+```
+PROJECT="${PROJECT}" \
+TRANSFERS_BUCKET="${BUCKET}" \
+GOOGLE_APPLICATION_CREDENTIALS="${HOME}/credentials.json" \
+ZONE=us-central1-b \
+PLATFORMS=debian-10 \
+go test -v ops_agent_test.go \
+ -test.parallel=1000 \
+ -tags=integration_test \
+ -timeout=4h
+```
+
+This will run the tests against the stable Ops Agent. To test against a pre-built
+but unreleased agent, you can use add the AGENT_PACKAGES_IN_GCS environment
+variable onto your command like this:
+
+```
+AGENT_PACKAGES_IN_GCS=gs://ops-agents-public-buckets-test-logs/prod/stackdriver_agents/testing/consumer/ops_agent/presubmit_github/debian/166/20220215-095636/agent_packages \
```

Comment:
[nit] I don't know where would be the best place to put this but it would be really useful to document instructions on how to find the prebuilt unreleased agent from presubmit runs.

Reply:
Done.